src/models/finetuning.py

```diff
@@ -0,0 +1,106 @@
+from transformers import AutoTokenizer, AutoModelForCausalLM
+from transformers import Trainer, TrainingArguments, TrainerCallback, DataCollatorForLanguageModeling
+import json
+import os
+import sys
+import pandas as pd
+import torch
+from torch.utils.data.dataset import Dataset
+import matplotlib.pyplot as plt
+import pickle
+
+current_dir = os.path.dirname(os.path.abspath(__file__))
+project_root = os.path.abspath(os.path.join(current_dir, os.pardir, os.pardir))
+sys.path.append(project_root)
+from src.data_processing.Formality_Transfer_Dataset import FormalityTransferDataset
+
+# Load model directly
+model_id = "gpt2-medium"
+tokenizer = AutoTokenizer.from_pretrained(model_id)
+
+tokenizer.add_special_tokens({
```

Comment:
have you considered loading the tokeniser that yoanna used?

Reply:
the loading_tokenization code does not pickle the tokenizer itself so I couldn't load it here but it would indeed be better to use the same tokenizer. However, here I'm basically making the same tokenizer so there should be no difference